include/state.hpp

```diff
@@ -4,6 +4,34 @@
 #include <vector>
 #include <memory>
 
+namespace states 
+{
+
+enum class StateId 
```

Comment:
- This should not be in the states namespace
- The enum values should be all lower case
- There is no need to append "STATE" to each enum value, since this is an enum class the values will be referred to as e.g. "StateId::trait"
- The enum values should be closer to the name of the corresponding Trait class (minus the "state" suffix which is redundant), e.g. "pick_trait" instead of "trait".